File: code/battery_vampire.c
/*
 * Infinite loop to calculate binary representation of 1/101
 */

#include <stdio.h>
#include <stdbool.h>

void binary_fraction(int numerator, int denominator) {
    int rem = numerator;  // Start with numerator 1
    int i = 0;
    
    printf("Binary representation of %d/%d: 0.", numerator, denominator);
    fflush(stdout);
    
    while (true) {
        rem = rem * 2;
        if (rem >= denominator) {
            printf("1");
            rem = rem - denominator;
        } else {
            printf("0");
        }
        
        // Optional: add some spacing for readability
        i++;
        if (i % 4 == 0) {
            printf(" ");
        }
        if (i % 80 == 0) {
            printf("\n");
        }
        
        fflush(stdout);
    }
}

int main() {
    binary_fraction(1, 101);
    return 0;
}

File: code/battery_vampire.swift
#!/usr/bin/env swift
/*
 * Infinite loop to calculate binary representation of 1/101
 */

import Darwin

func binaryFraction(numerator: Int = 1, denominator: Int = 101) {
    var rem = numerator  // Start with numerator 1
    print("Binary representation of \(numerator)/\(denominator): 0.", terminator: "")
    
    var i = 0
    while true {
        rem = rem * 2
        if rem >= denominator {
            print("1", terminator: "")
            rem = rem - denominator
        } else {
            print("0", terminator: "")
        }
        
        // Optional: add some spacing for readability
        i += 1
        if i % 4 == 0 {
            print(" ", terminator: "")
        }
        if i % 80 == 0 {
            print("\n", terminator: "")
        }
        
        // Force output flush
        fflush(stdout)
    }
}

// Main entry point
binaryFraction(numerator: 1, denominator: 101)
